Bug: https://github.com/rust-lang/rust/issues/57709

Index: src/tools/compiletest/src/main.rs
--- a/src/tools/compiletest/src/main.rs.orig
+++ b/src/tools/compiletest/src/main.rs
@@ -511,7 +511,11 @@ pub fn test_opts(config: &Config) -> test::TestOpts {
     test::TestOpts {
         filter: config.filter.clone(),
         filter_exact: config.filter_exact,
-        run_ignored: config.run_ignored,
+        run_ignored: if config.run_ignored {
+            test::RunIgnored::Yes
+        } else {
+            test::RunIgnored::No
+        },
         format: if config.quiet {
             test::OutputFormat::Terse
         } else {
