Author: Chris Coulson <chris.coulson@canonical.com>
Forwarded: TODO

--- a/src/bootstrap/test.rs
+++ b/src/bootstrap/test.rs
@@ -447,7 +447,7 @@
     }
 
     fn run(self, builder: &Builder) {
-        let rustdoc = builder.rustdoc(self.compiler.host);
+        let rustdoc = builder.out.join("bootstrap/debug/rustdoc");
         let mut cmd = builder.tool_cmd(Tool::RustdocTheme);
         cmd.arg(rustdoc.to_str().unwrap())
            .arg(builder.src.join("src/librustdoc/html/static/themes").to_str().unwrap())
