error[E0391]: cyclic dependency detected
  --> $DIR/issue-23302-1.rs:14:9
   |
LL |     A = X::A as isize, //~ ERROR E0391
   |         ^^^^^^^^^^^^^ cyclic reference
   |
note: the cycle begins when processing `X::A::{{initializer}}`...
  --> $DIR/issue-23302-1.rs:14:9
   |
LL |     A = X::A as isize, //~ ERROR E0391
   |         ^^^^^^^^^^^^^
   = note: ...which then again requires processing `X::A::{{initializer}}`, completing the cycle.

error: aborting due to previous error

For more information about this error, try `rustc --explain E0391`.
