Index: rustc/src/rustc/Cargo.toml
===================================================================
--- rustc.orig/src/rustc/Cargo.toml
+++ rustc/src/rustc/Cargo.toml
@@ -15,11 +15,3 @@ rustc_driver = { path = "../librustc_dri
 # Make sure rustc_codegen_ssa ends up in the sysroot, because this
 # crate is intended to be used by codegen backends, which may not be in-tree.
 rustc_codegen_ssa = { path = "../librustc_codegen_ssa" }
-
-[dependencies.jemalloc-sys]
-version = '0.1.8'
-optional = true
-features = ['unprefixed_malloc_on_supported_platforms']
-
-[features]
-jemalloc = ['jemalloc-sys']
