Description: Fix some debuginfo tests
Author: Kai Noda <nodakai@gmail.com>
Bug: https://github.com/rust-lang/rust/issues/40557
---
--- a/src/tools/compiletest/src/runtest.rs
+++ b/src/tools/compiletest/src/runtest.rs
@@ -645,6 +645,9 @@
                 script_str.push_str(&format!("file {}\n",
                                              exe_file.to_str().unwrap()
                                              .replace(r"\", r"\\")));
+                if self.config.gdb_native_rust {
+                    script_str.push_str("set language rust\n");
+                }
 
                 // Force GDB to print values in the Rust format.
                 if self.config.gdb_native_rust {
