Bug: https://github.com/rust-lang/rust/pull/47541
Applied-Upstream: commit:f4bcfc5317258c77e25ab5c0f3eec5599287fdbd
--- a/src/librustc_trans/cabi_sparc64.rs
+++ b/src/librustc_trans/cabi_sparc64.rs
@@ -50,7 +50,7 @@
     }
     let size = ret.layout.size;
     let bits = size.bits();
-    if bits <= 128 {
+    if bits <= 256 {
         let unit = if bits <= 8 {
             Reg::i8()
         } else if bits <= 16 {
@@ -84,6 +84,10 @@
     }
 
     let total = arg.layout.size;
+    if total.bits() > 128 {
+       arg.make_indirect();
+       return;
+    }
     arg.cast_to(Uniform {
         unit: Reg::i64(),
         total
