Index: rustc.git/src/test/run-make/no-duplicate-libs/main.rs
===================================================================
--- rustc.git.orig/src/test/run-make/no-duplicate-libs/main.rs
+++ rustc.git/src/test/run-make/no-duplicate-libs/main.rs
@@ -8,6 +8,8 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// ignore-test
+
 #[link(name = "foo")]
 #[link(name = "bar")]
 #[link(name = "foo")]
Index: rustc.git/src/test/run-make/no-duplicate-libs/Makefile
===================================================================
--- rustc.git.orig/src/test/run-make/no-duplicate-libs/Makefile
+++ rustc.git/src/test/run-make/no-duplicate-libs/Makefile
@@ -5,8 +5,8 @@ ifdef IS_MSVC
 all:
 else
 all:
-	$(RUSTC) foo.rs
-	$(RUSTC) bar.rs
-	$(RUSTC) main.rs
-	$(call RUN,main)
+#	$(RUSTC) foo.rs
+#	$(RUSTC) bar.rs
+#	$(RUSTC) main.rs
+#	$(call RUN,main)
 endif
