Description: Various fixes for Rust on MIPS
Author: Dragan Mladjenovic <dmladjenovic@wavecomp.com>
Bug-Debian: https://bugs.debian.org/881845
Last-Update: 2018-10-24

Index: rustc/src/librustc_codegen_llvm/llvm_util.rs
===================================================================
--- rustc.orig/src/librustc_codegen_llvm/llvm_util.rs
+++ rustc/src/librustc_codegen_llvm/llvm_util.rs
@@ -68,6 +68,9 @@ unsafe fn configure_llvm(sess: &Session)
         // during inlining. Unfortunately these may block other optimizations.
         add("-preserve-alignment-assumptions-during-inlining=false");
 
+    if sess.target.target.arch == "mips" ||
+        sess.target.target.arch == "mips64" { add("-fast-isel=0"); }
+
         for arg in &sess.opts.cg.llvm_args {
             add(&(*arg));
         }
